src/main/java/com/nutomic/syncthingandroid/activities/FolderActivity.java

```diff
@@ -360,6 +361,9 @@ private Dialog createDeleteDialog(){
                 .setMessage(R.string.remove_folder_confirm)
                 .setPositiveButton(android.R.string.yes, (dialogInterface, i) -> {
                     getApi().removeFolder(mFolder.id);
+                    //Remove saved data from share activity for this folder.
+                    SharedPreferences prefs = this.getSharedPreferences(ShareActivity.SHARED_PREFS_SAVED_DIRECTORIES, MODE_PRIVATE);
+                    prefs.edit().remove(mFolder.id).apply();
```

Comment:
it's probably better to handle this in RestApi.removeFolder(), in case that function is called from somewhere else. And you dont need to set MODE_PRIVATE, because that's the default.